src/main/java/com/google/devtools/build/lib/remote/GrpcRemoteCache.java

```diff
@@ -183,7 +237,10 @@ public void ensureInputsPresent(
     SettableFuture<Void> outerF = SettableFuture.create();
     bsAsyncStub()
         .read(
-            ReadRequest.newBuilder().setResourceName(resourceName).build(),
+            ReadRequest.newBuilder()
+                .setResourceName(resourceName)
+                .setReadOffset(offset)
+                .build(),
```

Comment:
Do I understand correctly that this will not work as is because you're requesting a different offset every time, but disregard this offset when writing to out, so you would need something like the code that you commented out below to do it right?

Reply:
You're completely right. Commented code is coming out, scope of Offset (and Hashing) has to move to creation outside of the retrier.